code/modules/projectiles/guns/ballistic/rifle.dm

```diff
@@ -67,11 +64,14 @@
 
 	SET_BASE_PIXEL(-8, 0)
 
+/obj/item/gun/ballistic/rifle/boltaction/add_bayonet_point()
+	AddComponent(/datum/component/bayonet_attachable, offset_x = 32, offset_y = 12)
+
 /obj/item/gun/ballistic/rifle/boltaction/sawoff(mob/user)
 	. = ..()
 	if(.)
 		spread = 36
-		can_bayonet = FALSE
+		qdel(GetComponent(/datum/component/bayonet_attachable))
```

Comment:
For bonus points you can make this a signal, `COMSIG_GUN_SAWN_OFF`, which deletes the bayonet point.